src/controllers/accounts/AccountsConvertController.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ApiPromise } from '@polkadot/api';
+import { KeypairType } from '@polkadot/util-crypto/types';
+import { RequestHandler } from 'express';
+import { BadRequest } from 'http-errors';
+
+import { AccountsConvertService } from '../../services/accounts';
+import AbstractController from '../AbstractController';
+
+export default class AccountsConvertController extends AbstractController<AccountsConvertService> {
+	constructor(api: ApiPromise) {
+		super(api, '/accounts/:address/convert', new AccountsConvertService(api));
+		this.initRoutes();
+	}
+
+	protected initRoutes(): void {
+		this.safeMountAsyncGetHandlers([['', this.accountConvert]]);
+	}
+
+	private accountConvert: RequestHandler = (
+		{ params: { address }, query: { scheme, prefix, publicKey } },
+		res
+	) => {
+		// Validation of the `scheme` query param
+		const cryptoScheme = (
```

Comment:
We can clean this up with some more specific type level validation. 

For example: we can set the `RequestHandler` to accept types we create.

Ex: 
```typescript
interface IConvertQueryParams extends Query {
    scheme: string; // You might be able to set this to a union type of `'ed25519' | 'sr25519' | 'ecdsa'`
    prefix: string;
    publicKey: string;
}

// Then you would just have RequestHandler type be like
RequestHandler<unknown, unknown, unknown, IConvertQueryParams>
```

Please use https://github.com/paritytech/substrate-api-sidecar/blob/master/src/types/requests.ts#L49 as guidance

Reply:
This will also clean up the type checks below.

Reply:
Thank you @TarikGul for your comments and suggestions! Very much appreciated!

I tried the solution of defining the query params types in `requests.ts` but I can not see the value of this change very clearly since I still need to keep all of my validation code in the controller. Maybe I just implemented in a wrong way so let me explain below what I tried to do.

I added the `interface` you suggested (the `union` type does not seem to have any effect in runtime so I just set the types to `string`). Adding this, resulted in the following changes in my code : 

- I replaced wherever I had `typeof` with `undefined`, e.g.
    changed this check 
    ```
    const networkPrefix = typeof prefix !== 'string' ? '42' : prefix;
    ```
    to this
    ```
    const networkPrefix = prefix === undefined ? '42' : prefix;
    ```
- The above point is the same for the other 2 query params
- However, I still need to check if the values given in the query params are valid, e.g. valid values for `scheme` should be `'ed25519' | 'sr25519' | 'ecdsa'` and if they are not, throw an error. If I do not have a check for this in the `controller` then it throws when it goes to the `service` and there the error is 
  ```
  "Expected a keyring type of either 'ed25519', 'sr25519', 'ethereum' or 'ecdsa', found 'domi"
  ```
  which is not what I would like to convey to the user.
- The same for the values of the `publicKey`. I would like to throw an error and tell the user to give a value of `true` or `false` instead of not throwing any errors and just set `publicKey` to `false`.

If the validation checks in the `AccountsConvertController.ts` make it confusing then maybe I could add a middleware for this ? Like it is implemented in [validateAddressMiddleware.ts](https://github.com/paritytech/substrate-api-sidecar/blob/master/src/middleware/validate/validateAddressMiddleware.ts) ?